src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Value.cs

```diff
@@ -0,0 +1,264 @@
+                            {
+                                info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, platformName, version, negated: false);
+                                return true;
+                            }
+                            else if (TryDecodeOSVersion(arguments, valueContentAnalysisResult, out version, 1))
+                            {
+                                // OperatingSystem.IsOSPlatformVersionAtLeast(string platform, int major, int minor = 0, int build = 0, int revision = 0)
+                                Debug.Assert(invokedPlatformCheckMethod.Name == IsOSPlatformVersionAtLeast);
+                                info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, literal.ConstantValue.Value.ToString(), version, negated: false);
+                                return true;
+                            }
+                        }
+                        else if (literal.Type?.SpecialType == SpecialType.System_Int32)
+                        {
+                            // Accelerators like OperatingSystem.IsPlatformNameVersionAtLeast(int major, int minor = 0, int build = 0, int revision = 0)
+                            var platformName = SwitchVersionedPlatformName(invokedPlatformCheckMethod.Name);
+
+                            if (platformName != null && TryDecodeOSVersion(arguments, valueContentAnalysisResult, out var version))
+                            {
+                                info = new RuntimeMethodValue(invokedPlatformCheckMethod.Name, platformName, version, negated: false);
+                                return true;
+                            }
+                        }
+                    }
+                }
+
+                info = default;
+                return false;
+            }
+
+            private static string? SwitchVersionedPlatformName(string methodName)
+            => methodName switch
+            {
+                IsWindowsVersionAtLeast => Windows,
+                IsMacOSVersionAtLeast => MacOS,
+                IsIOSVersionAtLeast => IOS,
+                IsAndroidVersionAtLeast => Android,
+                IsFreeBSDVersionAtLeast => FreeBSD,
+                IsTvOSVersionAtLeast => TvOS,
+                IsWatchOSVersionAtLeast => WatchOS,
+                _ => null
+            };
+
+            private static string? SwitchPlatformName(string methodName)
+                 => methodName switch
+                 {
+                     IsWindows => Windows,
+                     IsLinux => Linux,
+                     IsMacOS => MacOS,
+                     IsIOS => IOS,
+                     IsBrowser => Browser,
+                     IsAndroid => Android,
+                     IsFreeBSD => FreeBSD,
+                     IsTvOS => TvOS,
+                     IsWatchOS => WatchOS,
+                     _ => null
+                 };
```

Comment:
Let's not hard code the platforms. Instead, I suggest an implementation like this:

```C#
private static string? SwitchPlatformName(string methodName)
{
    if (!methodName.StartsWith("Is", StringComparison.Ordinal))
        return null;

    const string optionalSuffix = "VersionAtLeast";
    if (methodName.StartsWith(optionalSuffix, StringComparison.Ordinal))
        methodName = methodName.Substring(0, methodName.Length - optionalSuffix.Length);

    return methodName.Substring(2);
}
```

Reply:
We've gone back and forth on this. For 5.0, we need to leave it hard-coded (out of time). But we can apply this approach after release.